react/src/components/FolderExplorerHeader.tsx

```diff
@@ -105,10 +128,154 @@ const FolderExplorerHeader: React.FC<FolderExplorerHeaderProps> = ({
                     preview={false}
                   />
                 }
-                onClick={() =>
-                  // @ts-ignore
-                  folderExplorerRef.current?._executeFileBrowser()
-                }
+                loading={isLoadingToStartFileBrowser}
+                onClick={() => {
```

Comment:
Let's use `BAIButton` with the `action` prop. Once we apply it, managing the `isLoadingToStartFileBrowser` state will no longer be necessary. Please refer to the `startSession` implementation in `SessionLauncher` for reference.